### ISSUE TITLE
[Bug]: Push doesnt work on mobile

### ISSUE THREAD

**Issue by MakiRollDev** [denolehov/obsidian-git#653]
### Describe the bug

So I had to copy my repository onto my phone by hand (via USB on my PC). 
The problem is that when I try to push my changes to Github, it says "MissingParameterError: The function requires a 'remote or URL''"

I have tried:
- creating a new remote on my phone
- switching to remote branch
- appending '.git' to the repo URL
- 

### Steps to reproduce

I do not have the hardware to test this on other devices
- Set up obsidian git on desktop
- Copy it over to your android device and open it in obsidian
- set up authentication to github (Name and Token)
- commit all changes
- push

### Operating system

Android

### Installation Method

None

### Plugin version

2.22.0

**Comment by kaisersakhi** [comment]
@MakiRollDev Did you find any workaround?

**Comment by MakiRollDev** [comment]
@kaisersakhi No sadly not I just switched to Google drive

**Comment by trinityzhub** [comment]
Same error for me in my  Android Phone 
```
MissingParameterError : The Function requires a "remote OR url" parameter . But none was provided
```

**Comment by Irvus** [comment]
same problem. I would like to give some logs or more info, but don't know how

**Comment by Irvus** [comment]
no, i no idea how to open dev tools from android. ok, android 11 rkq1.200826.002, MIUI 12.5.6, pull works correctly. field "username on your git server does no effect, i changed it to completly incorrect, and all worked after reboot of the phone. Also, the new token i generated in github was never used. There are files that were modifed after last commin inside .git of the phone, some /objects, index, /refs/heads and so on

Also I add .git/config files from windows and android. As I see, username and psw fields do nothing because url was not modified after copying from win, but author name and email were changed. Maybe that's okay for git security, i'm bad in it
[config-windows.txt](https://github.com/user-attachments/files/17941283/config-windows.txt)
[config-android.txt](https://github.com/user-attachments/files/17941284/config-android.txt)


Now something I was doing: i used [this guide](https://www.reddit.com/r/ObsidianMD/comments/17odzjb/obsidian_android_syncing_via_github_in_2023/) to sync my devices

>  Step 5: Add the folder as a vault in Obsidian
> 
>     1. Open Obsidian
> 
>     2. Tap the sidebar icon in the top left corner
> 
>     3. Click the vault name in the top left corner, click "Manage valuts..."
> 
>     4. Click "Open folder as vault" and navigate the the folder you copied over
> 
>     5. Obsidian Git should try to fetch
>     (or open command pallette by swiping down -> "Obsidian Git: Pull")
> 
>     6. Enter your username & access token when Obsidian git asks for it
>     (using your GitHub password might work, but I did not try it)
> 
>     7. Congratulations, you now have working sync!

At 6. there was an error which was solved by filling the author email field. After that pull worked correctly

and thats why: i added the .git/config after i click "open folder as vault" - there is no email now
[config_after Open folder as vault.txt](https://github.com/user-attachments/files/17941366/config_after.Open.folder.as.vault.txt)

I tried to change the access key to classic and finegrained tokens, but this made no difference
![изображение](https://github.com/user-attachments/assets/66886bd7-b6f9-481e-969f-17ecaa1a6916)

here is with incorrect github username (github.com/Irvu/main-obsidian-vault.git)
![изображение](https://github.com/user-attachments/assets/32331e56-1df7-43d2-975b-863e782bf0a8)

and here with edited token (url = https://github_pat_21... , 1 changed to 2)

Then I regenerated all tokens, saw a menu with login/password or token. I've filled it with my login and token, but as i know, .git/config rewrites global config. But that's not right in this case, because i broke the .git/config earlier (token is github_pat_1 now)
[config broken.txt](https://github.com/user-attachments/files/17950122/config.broken.txt)

i reinstalled the app, but it made no difference, the mistake is still here

then i removed config file from .git, and this is new generated
[config andr-obs made itself.txt](https://github.com/user-attachments/files/17950358/config.andr-obs.made.itself.txt)

so, the mistake
![изображение](https://github.com/user-attachments/assets/0c50b4ca-2d78-4d5e-bd1f-0fc8705d8841)
or with writing the exact link in the field of loging https://ghp_Uandsoon@github.com/Irvus/main-obsidian-vault.git
![изображение](https://github.com/user-attachments/assets/dab325e2-c200-47bc-8a12-fd8802154b67)
![изображение](https://github.com/user-attachments/assets/fd13f7cf-ea9a-4741-be17-0551521dc999)

**Comment by firedev** [comment]
I couldn't fix the issue from scratch but here is a workaround. For iPhone, I airdropped the whole vault folder. Moved it to `iPhone/Obsidian` in `Files`. Then opened it as a new vault. Git repo was set to `https://token@github.com/myname/repo`. This way it started to work.

**Comment by marcnesium** [comment]
Same issue here on Android 15. Fetch and Pull works, but push results in "The function requires a "remote OR url" parameter but none was provided.
My obsidian vault was copied from PC onto the android phone as suggested.
Could find any working workaround yet.
If needed i could provide logs.
Side note: i use gitea as remote repo.

**Comment by marcnesium** [comment]
Ah, i just found a **workaround** (that is perhaps a hint for a bugfix)!

My remote ws not called "origin" but "gitea".
With a texteditor i changed in the .gut/config file 
[remote "gitea"]
remote = gitea 
to
[remote "origin"] 
remote = origin

That works for me.

**Comment by Ortega-Dan** [comment]
I was having this issue today in Android, and It ended up fixed after adding the "Commit Statuses" to my Fine-Grained Github PAT (which I know the instructions clearly state, but I somehow only provided Contents(code) and Metadata first) but it needs to be in the following way:

![Image](https://github.com/user-attachments/assets/fcc1eb1e-b580-4695-ab3c-8dc4e2c2fb73)

The other problem is that it was a brand new repo, so when it asked me to select the or type the branch name, for it I was typing "main" but I somehow realized it wanted the "fully qualified" branch name, being "origin/main" (in my case, instead of plain "main"). So it means, if your remote shows as "whatever", it should be "whatever/thebranchyouwant".

After these two adjustments, everything worked well. 👍

**Comment by mbianchidev** [comment]
Quickest solution to correctly sync any vault for Android users.

0. Commit/push everything you have on the vault you want to sync
1. Create an empty vault on mobile
2. Download the community plugin on mobile and then close your Obsidian Android app
3. Connect your phone to your laptop (where you already use Obsidian with a git synce)
4. Copy the entire repo to your mobile vault location, including the .git and .obsidian folders
5. Re-open the app - PROFIT

It's a one time only thing, after that push, fetch and whatever always works.
Repeat the process for each of your vaults if needed.

**Comment by TheDeafOne** [comment]
> Ah, i just found a **workaround** (that is perhaps a hint for a bugfix)!
> 
> My remote ws not called "origin" but "gitea".
> With a texteditor i changed in the .gut/config file 
> [remote "gitea"]
> remote = gitea 
> to
> [remote "origin"] 
> remote = origin
> 
> That works for me.

My issue was also caused by an incorrect remote reference. For some reason the toml tag was correct, but not the actual value assignment (was remote = .). Used a-Shell to edit this to origin, and everything began working again.

**Comment by TLS15** [comment]
I was naming my remote, but I kept getting this error. The only valid name was origin. If you name your upstream like this it should work.
